analyzers/tests/SonarAnalyzer.UnitTest/TestCases/GetTypeWithIsAssignableFrom.CSharp9.cs

```diff
@@ -1,17 +1,20 @@
 ﻿Apple apple = new();
 Fruit f = apple;
 
-var b = f is Apple; // Compliant
-b = f is not Apple; // Compliant
-b = apple.GetType() == typeof(int?); // Compliant
-b = apple.GetType().IsInstanceOfType(f.GetType()); // Compliant
-b = f as Apple == null; // Noncompliant
-b = f as Apple is null; // FN
-b = f as Apple is not null; // FN
-b = true && (apple) is Apple; // Noncompliant
-b = !(apple is Apple); // Noncompliant
+var b = f is Apple;     // Compliant
+b = f is not Apple;     // Compliant
+b = apple.GetType() == typeof(int?);                // Compliant
+b = apple.GetType().IsInstanceOfType(f.GetType());  // Compliant
+b = f as Apple == null;         // Noncompliant
+b = f as Apple is null;         // Noncompliant
+b = f as Apple is not null;     // Noncompliant
+b = f as Apple is not not not not null; // Noncompliant
+b = f as Apple is Apple { Size: 42 };   // Compliant
+b = f as Apple is Apple;        // Noncompliant
+b = true && (apple) is Apple;   // Noncompliant
+b = !(apple is Apple);          // Noncompliant
 
-if (apple is { }) { } // Compliant
+if (apple is { }) { }           // Compliant
```

Comment:
could you also add more pattern types to make sure it doesn't break the logic or behave strange?

```
public class C {
    public void M(Apple a) {
        if (a is ("Sweet", "Red"))
        {
        }
        if (a is ("Sweet", "Red"))
        {
        }
        if (a is {Taste: "Sweet", Color: "Red"})
        {
        }
    }
}

public class Apple {
    public string Taste;
    public string Color;
    public void Deconstruct(out string x, out string y) => (x, y) = (Taste, Color);
}
```